packages/appium/docs/en/ecosystem/build-drivers.md

```diff
@@ -2,4 +2,617 @@
 title: Building Appium Drivers
 ---
 
-TODO
+Appium wants to make it easy for anyone to develop their own automation drivers as part of the
+Appium ecosystem. This guide will explain what's involved and how you can accomplish various driver
+development tasks using the tools Appium provides. This guide assumes you (1) are a competent user of
+Appium, (2) are a competent Node.js developer, and (3) that you have read and understood the
+[Driver Intro](../intro/drivers.md).
+
+If that describes you, great! This guide will get you started.
+
+## Before you create your driver
+
+Before you get to work implementing your driver, it's important to have a few things sorted out.
+For example, you need to know what your driver will do. Which platform is it trying to expose
+WebDriver automation for?
+
+Appium doesn't magically give you the power to automate any platform. All it does is give you a set
+of convenient tools to make implementing the WebDriver Protocol easy. So if you want to create
+a driver for some new app platform, let's say, you'll need to already know how you would automate
+apps on that platform *without Appium*.
+
+This usually means that you need to be very familiar with app development for a given platform. And
+it usually means that you will rely on tools or SDKs provided by the platform vendor.
+
+Basically, if you can't answer the question "how would I launch, remotely trigger behaviours, and
+read state from an app on this platform?" then you're not quite ready to write an Appium driver.
+Make sure you do the research to feel comfortable that there *is* a path forward. Once there is,
+coding it up and making it available as an Appium driver should be the easy part!
+
+## Other drivers to reference
+
+One of the greatest things about building an Appium driver is that there are already a number of
+open source Appium drivers which you can look at for reference. There is
+a [fake-driver](https://github.com/appium/appium/tree/2.0/packages/fake-driver) sample driver which
+does basically nothing other than showcase some of the things described in this guide.
+
+And of course, all of Appium's official drivers are open source and available in repositories at
+the project's GitHub organization. So if you ever find yourself asking, "how does a driver do X?",
+read the code for these drivers! Also don't be afraid to ask questions of the Appium developers if
+you get stuck; we're always happy to help make sure the driver development experience is a good
+one!
+
+## Basic requirements for Appium drivers
+
+These are the things your driver *must* do (or be), if you want it to be a valid Appium driver.
+
+### Node.js package with an Appium manifest
+
+All Appium drivers are fundamentally Node.js packages, and therefore must have a valid
+`package.json`. Your driver is not limited to Node.js, but whatever you do with your driver must
+ultimately be triggered by Node.js.
```

Comment:
```suggestion
`package.json`. Your driver is not _limited_ to Node.js, but it must provide an adapter written in Node.js so it can be loaded by Appium.
```